soundsgood.dsp

```diff
@@ -337,42 +337,89 @@ mscomp8i(target) =
   : ms_dec
 ;
 
-
-Eight_band_Compressor_N_chan(N) =
-   crossover
-  : compressors
-  : mixer
+Eight_band_Compressor_N_chan(N) = si.bus (N) <: si.bus (2 * N)
+  : (calcredux(N,(flist)), si.bus (N))
+  : groupGainOut
+  : par(i, N, shelfcascade ((flist)))
   : outputGain
 with {
-  inputGain = par(i, N, _*inGain);
-  crossover =
-    (
-      (crossoverFreqs<:par(i, Nr_crossoverFreqs, _<:si.bus(N)))
-     ,si.bus(N)
-    )
-    <: ro.interleave(N,Nr_bands)
-    :  par(i, N, fi.crossover8LR4)
-    :  ro.interleave(Nr_bands,N);
-  compressors =
-    (strength_array , thresh_array , att_array , rel_array , knee_array , link_array, ro.interleave(N,Nr_bands))
+
+  /* re-order, interleave output:  left gain [Nr_bands .. 1], left-signal , right gain [Nr_bands .. 1], right-signal
+   * Note: shelfcascade takes gains in reverse order */
+  groupGainOut  =
+    (ro.interleave(Nr_bands,N), si.bus(N))
+    : (ro.interleave(N, Nr_bands+1))
+    : par(i, N, ro.cross(Nr_bands),_);
+
+  /* ******* 8< *******/
+  // TODO: use co.peak_compression_gain_N_chan_db when it arrives in the current faust version
+  peak_compression_gain_mono_db(strength,thresh,att,rel,knee,prePost) =
+    abs : ba.bypass1(prePost,si.onePoleSwitching(att,rel)) : ba.linear2db : gain_computer(strength,thresh,knee) : ba.bypass1((prePost !=1),si.onePoleSwitching(rel,att))
+  with {
+    gain_computer(strength,thresh,knee,level) =
+      select3((level>(thresh-(knee/2)))+(level>(thresh+(knee/2))),
+        0,
+        ((level-thresh+(knee/2)) : pow(2)/(2*max(ma.EPSILON,knee))),
+        (level-thresh))
+      : max(0)*-strength;
+  };
+
+  peak_compression_gain_N_chan_db(strength,thresh,att,rel,knee,prePost,link,1) =
+    peak_compression_gain_mono_db(strength,thresh,att,rel,knee,prePost);
+
+  peak_compression_gain_N_chan_db(strength,thresh,att,rel,knee,prePost,link,N) =
+    par(i, N, peak_compression_gain_mono_db(strength,thresh,att,rel,knee,prePost))
+    <: (si.bus(N),(ba.parallelMin(N) <: si.bus(N))) : ro.interleave(N,2) : par(i,N,(it.interpolate_linear(link)));
+
+  /* ******* >8 *******/
+
+  compressor(N,prePost,strength,thresh,att,rel,knee,link) = peak_compression_gain_N_chan_db (strength,thresh,att,rel,knee,prePost,link,N);
+
+  comps(N) = (strength_array, thresh_array, att_array, rel_array, knee_array, link_array, ro.interleave (N,Nr_bands))
     : ro.interleave(Nr_bands,6+N)
-    : par(i, Nr_bands, compressor(meter(i+1),N,prePost)) ;
-  mixer = si.bus(N*Nr_bands):>si.bus(N);
+    : par(i, Nr_bands, compressor(N,prePost)) // : si.bus (N * Nr_bands)
+    : par(c, N, par(b, Nr_bands, meter(c+1, b+1)));
```

Comment:
I think you want:
```faust
: par(b, Nr_bands, par(c, N, ba.linear2db : meter(b+1, c+1)));
```
That way you get 8 times MS, like in the old version.

Reply:
That'd be incorrect order. the gain parameters are order 
  8 values for  leftchannel,  8 values for right channel
hence  `par(c, N, par(b, Nr_bands, meter(c+1, b+1))); `